src/Chainweb/Rosetta/Internal.hs

```diff
@@ -351,66 +395,94 @@ nonGenesisTransaction logs initial rest target
     fromShortCircuit (Left (Right tx)) = pure (Just tx)
         -- Tx found
 
-
 -------------------------
 -- Remediation Helpers --
 -------------------------
 
-remediationRequestKey :: RequestKey
-remediationRequestKey = RequestKey $ pactHash "remediation"
-
-
--- | Group the rest of the logs into a single transaction id because
--- remediations all have different txIds and we don't have access to
--- their command results.
--- NOTE: If a normal transaction occurs in this block, it will be grouped with
--- the remediation changes.
-getRemediationsTx
-    :: [(TxId, [AccountLog])]
-    -> Transaction
-getRemediationsTx logsList =
-  let t = rkToTransactionId remediationRequestKey
-      f (tid, ali) = map (operation Successful TransferOrCreateAcct tid) ali
-      ops = indexedOperations $ concat $ map f logsList
-  in (Transaction t ops Nothing)
-
-
--- TODO: Do all the remediations touch the coin contract? This might help
--- with connecting a remediation Command with a TxId log.
--- TODO: Are we loading "pact/coin-contract/v2/load-fungible-asset-v2.yaml"
--- "pact/coin-contract/v2/load-coin-contract-v2.yaml" every time a remediation is
--- run? Is it always 3 commands that are run for each remediation?
+-- | Given a remediation Command and its assumed TxId, tries to get its coin table logs.
+--   If the last coin table log TxId it sees matches the remediation's TxId, then assumes
+--   that this log corresponds to that remediation. Otherwise, a rosetta transaction is created
+--   for that remediation with no coin table logs.
+remediationAcc
+    :: AccumulatorType (TxAccumulator rosettaTx)
+    -> TxAccumulator rosettaTx
+    -> (Command payload, TxId)
+    -> TxAccumulator rosettaTx
+remediationAcc accTyp acc (remTx, remTid) =
+  case (_txAccumulator_logsLeft acc) of
+    (logTid,logs):rest
+      | logTid == remTid -> -- remediation touched coin table
+        let ops = indexedOperations $!
+              UnindexedOperations
+              { _unindexedOperation_fundOps = []
+              , _unindexedOperation_transferOps = makeOps (logTid, logs)
+              , _unindexedOperation_gasOps = []
+              }
+            rosettaTx = rosettaTransactionFromCmd remTx $! ops
+        in makeAcc rest rosettaTx
+    rest -> -- list of logs empty or remediation didn't touch coin table
+      makeAcc rest $!
+      rosettaTransactionFromCmd remTx []
+  where
+    makeAcc restLogs rosettaTx =
+      accumulatorFunction accTyp acc restLogs rosettaTx
+    makeOps (tid, logs) =
+      map (operation Remediation TransferOrCreateAcct tid) logs
+
+-- | Matches all transactions in a remediation block (including coinbase, remediations,
+--   and user transactions) to their coin table logs.
+--  Matches coinbase logs first, then remediations (assumes that each remediation transaction
+--  incremented the TxId counter), and uses same algorithm as non-genesis transaction matching
```

Comment:
Is this assumption correct? Under the hood, the remediations use the applyGenesis helper function from TransactionExec.

Reply:
Resolving this comment as it's a duplicate of the conversation occurring in https://github.com/kadena-io/chainweb-node/pull/1135#discussion_r798060937